### PR TITLE
Update Nim bindgen (May 2022)

### PR DESCRIPTION
**Pull request by gustavolsson**
* Follow nim code style
* Avoid keyword clashes
* Remove redundant functions, refactor
* Use nim bitfields for some consts
* Generate bindings for sokol_glue.h (I _think_ this makes it possible to use gfx without app)

Corresponding PR for sokol-nim: https://github.com/floooh/sokol-nim/pull/12

### CODE REVIEW

File: bindgen/gen_nim.py [hunk 1]
```diff
@@ -252,239 +298,141 @@ def extract_ptr_type(s):
...
         for value in sorted(item_names_by_value):
             name = item_names_by_value[value]
-            l(f"  {name} = {value},")
+            l(f"    {name} = {value},")
     else:
         for name in item_names_by_value.values():
-            l(f"  {name},")
+            l(f"    {name},")
+    if bitfield is not None:
+        l(f"  {enum_name_nim}s = set[{enum_name_nim}]")
```
Comment (gustavolsson) [inline bindgen/gen_nim.py:428]: Hmm.. this doesn't seem to work. I thought Nim bitfields were interoperable with C..

Comment (gustavolsson) [inline bindgen/gen_nim.py:428]: Hmm.. this doesn't seem to work. I thought Nim bitfields were interoperable with C..

**Comment by floooh** [comment]
Random thought: I wonder if it makes sense to get rid of the glm dependency? In sokol-zig I just wrote a simple math module with just the stuff that's needed (mostly ripped from HandmadeMath.h):

https://github.com/floooh/sokol-zig/blob/master/src/examples/math.zig

**Comment by garettbass** [comment]
I’ve switched to Zig.  No objections from me :)

**Comment by garettbass** [comment]
Have you seen Zig’s `@Vector()`?

**Comment by floooh** [comment]
@garettbass I had tinkered a bit with @Vector in my pacman.zig project, but replaced it with a regular struct because there were some miscompilation issues at the time (see: https://github.com/floooh/pacman.zig/pull/9)

**Comment by gustavolsson** [comment]
> @gustavolsson I'm planning to look into this next, is the PR ready or do you plan any more big changes?
> 
> Also, @garettbass any objections? Do you currently have anything important based on your last changes to the nim bindings?

Nope, all done :)

I wanted to use Nim bitfields for flag enums, bit they don't seem to be compiled to plain integers so it didn't work out. I will create a separate PR if I figure it out!

**Comment by floooh** [comment]
Just a heads up, I will spend a bit more time with the Nim bindings and language bindings in general in the next few days to weeks:

- update the language bindings automatically on commits/merges to the main branch via GH Actions (wip here: https://github.com/floooh/sokol/blob/master/.github/workflows/gen_bindings.yml)
- add Nim output to the sokol-shdc tool (similar to the Zig output mode: https://github.com/floooh/sokol-tools/blob/master/src/shdc/sokolzig.cc)
- ...and I will experiment a bit to restructure the "application model" when using the Nim bindings, I want to find out if it makes more sense to use sapp.run() instead of a sokol_main() function, similar to how it works in Zig (https://github.com/floooh/sokol-zig/blob/3fcc60c1c8be8fee1d5e51f9d76d5b1886725056/src/examples/clear.zig#L31-L43), if this works it would also get rid of the extra ```--noMain``` flag

**Comment by gustavolsson** [comment]
Sounds great! :)

**Pull request merged**